warehouse/query-core/src/main/java/datawave/query/planner/FederatedQueryPlanner.java

```diff
@@ -0,0 +1,100 @@
+package datawave.query.planner;
+
+import datawave.query.CloseableIterable;
+import datawave.query.config.ShardQueryConfiguration;
+import datawave.query.exceptions.DatawaveQueryException;
+import datawave.query.planner.pushdown.rules.PushDownRule;
+import datawave.query.tables.CountingShardQueryLogic;
+import datawave.query.tables.ScannerFactory;
+import datawave.query.tables.ShardQueryLogic;
+import datawave.query.util.MetadataHelper;
+import datawave.webservice.common.logging.ThreadConfigurableLogger;
+import datawave.webservice.query.Query;
+import datawave.webservice.query.configuration.GenericQueryConfiguration;
+import datawave.webservice.query.configuration.QueryData;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.apache.commons.jexl2.parser.ASTJexlScript;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+
+import java.util.Collection;
+
+public class FederatedQueryPlanner extends QueryPlanner implements Cloneable {
+    
+    QueryPlanner originalQueryPlanner;
+    ShardQueryLogic originalShardedQueryLogic;
+    CountingShardQueryLogic originalcountingShardQueryLogic;
+    
+    private static final Logger log = ThreadConfigurableLogger.getLogger(FederatedQueryPlanner.class);
+    {
+        log.setLevel(Level.DEBUG);
+    }
+    
+    public FederatedQueryPlanner(QueryPlanner original, ShardQueryLogic logic) {
+        // originalQueryPlanner.metadataHelper.getAllFieldMetadataHelper().getIndexDates();
+        originalQueryPlanner = original;
+        originalShardedQueryLogic = logic;
+    }
+    
+    @Override
+    public CloseableIterable<QueryData> process(GenericQueryConfiguration config, String query, Query settings, ScannerFactory scannerFactory)
+                    throws DatawaveQueryException {
+        return null;
```

Comment:
I thought maybe you would have more to show here....

Reply:
Got it @ivakegg.  Its the process function in the FederatedQueryPlanner that I start the implementation.   I am glad I have the correct class hierarchy and overrides correct.  I am trying to harmonize the new FederatedQueryPlanner with the present test framework and query framework  so they can run and be tested side by side before I rip the original framework out.  That way I can compile with tests and see if I am breaking anything.   Thanks for kicking me off in the right direction.